src/org/traccar/protocol/AdmProtocolDecoder.java

```diff
@@ -119,4 +123,41 @@ protected Object decode(
         return null;
     }
 
+    private Position parseCommandResponse(Channel channel, SocketAddress remoteAddress, ChannelBuffer buf) {
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress);
+        if (deviceSession == null) {
+            return null;
+        }
+
+        Position position = new Position();
+        position.setProtocol(getProtocolName());
+        position.setDeviceId(deviceSession.getDeviceId());
+
+        getLastLocation(position, null);
+
+        position.set(Position.KEY_RESULT, buf.readBytes(129).toString(StandardCharsets.UTF_8));
+
+        return position;
+    }
+
+    @Override
+    protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg) throws Exception {
+        ChannelBuffer buf = (ChannelBuffer) msg;
+
+        buf.readUnsignedShort(); // device id
+
+        int size = buf.readUnsignedByte();
+        if (size != CMD_RESPONSE_SIZE) {
```

Comment:
Is there a better way to identify command response?

Reply:
No. Command response identified only by packet size